start.py

```diff
@@ -164,6 +164,11 @@ def cm_server_not_dead(primary_node):
                       if parcel.product == 'CDH' and parcel.stage in ('ACTIVATING',
                                                                       'ACTIVATED'))
 
+    # Wait on all slave nodes to get back to CM
+    def cm_all_slaves_ready(expected):
+      return len(deployment.get_all_hosts()) == expected
+    wait_for_condition(cm_all_slaves_ready, [len(nodes)])
```

Comment:
Can shorten things by using:
```
wait_for_condition(lambda: len(deployment.get_all_hosts()) == len(nodes))
```

At some point, I'm gonna go through and do the same for some of the trivial functions I wrote here because this file is getting silly long.

Reply:
Done.